### ISSUE TITLE
Icons move around after customize and restart

### ISSUE THREAD

**Issue by Rmano** [xournalpp/xournalpp#1805]
**Affects versions:**
 - Linux Ubuntu 18.04, standard gnome desktop
 - Version of Xournal++: master branch 1.1.0+dev cb964dc4
 - Installation method:compiled

**Describe the bug**
I modify the position of a copy of "right hand note taking" toolbar (BTW, nice!) like this (notice the arrows signaling gadgets and dividing lines):

![2020-03-27_17-00](https://user-images.githubusercontent.com/6414907/77776006-a74f3480-704d-11ea-94c6-5c06c9236a9b.png)

I save, exit, and on restart, I have this:

![2020-03-27_17-01](https://user-images.githubusercontent.com/6414907/77776043-b59d5080-704d-11ea-90a6-a2c33d9c1ad0.png)

As you can see, cheese ;-) has moved around.

**To Reproduce**
1. Create a copy of "Right hand taking notes" profile 
2. Customize it as shown in the first screenshot
3. Exit `xournalpp` and start it again

**Expected behavior**

The icons stay put ;-)

**Comment by superenginegit** [comment]
I have looked into this issue as I found it to be quite annoying. 
And it seems to originate from `ToolbarAdapter::toolbarDragDataReceivedCb`
Especially this line: `gint pos = gtk_toolbar_get_drop_index(toolbar, wx, wy);`
The Position is sometimes off by 1, so an icon drifts to the right one slot. The item this does happen with the most is the separator, which is almost always. That can be fixed by 
inserting the separator with this line `int newId = tb->insertItem(name, "SEPARATOR", pos-1);`
This fix makes it behave as expected in about 90% of all cases. The other items don't have such an easy workaround, as the issue only occurs seldomly. 

I see two usable workarounds for now:
1. A user can try just inserting the item that is misplaced one slot to the left
2. The config file `toolbar.ini` can be edited by hand to correct misplaced items
